Add 'Handling of entities with identical IDs' section

marcusrein opened the pull request.

leoyvens commented:
Have you tested this? I think there is a nuance where if the field is explicitly set to `null`, it will be removed in the entity. But if it is simply omited, then it won't affect the enitty.

schmidsi commented:
I'd add this to "[Writing Mappings](https://thegraph.com/docs/en/developing/creating-a-subgraph/#writing-mappings)" instead of "Defining Entities"

marcusrein commented:
I tested the `null` handling within a subgraph and updated the description accordingly @leoyvens.

I moved the section to "Writing Mappings" @schmidsi

leoyvens commented on website/pages/en/developing/creating-a-subgraph.mdx:
```diff
@@ -537,6 +533,12 @@ Every entity has to have an `id` that is unique among all entities of the same t
+When creating and saving a new entity, if an entity with the same ID already exists, the properties of the new entity are preferred during the merge process. This means that the existing entity will be updated with the values from the new entity.
+
+If a null value is set for a field in the new entity with the same ID, the existing entity will be updated with the null value.
+
```
What if no value is set at all? Good to be explicit about that case.

leoyvens replied:
huh, surprising to me, good that we tested!

marcusrein commented:
I added one more paragraph considering this handling @leoyvens, let me know if you approve, thank you.

leoyvens reviewed (approved):
lgtm
